src/zepben/evolve/model/cim/iec61970/base/core/create_core_components.py

```diff
@@ -0,0 +1,173 @@
+    """
+    BaseVoltage(IdentifiedObject)
+    IdentifiedObject: mrid, name, description, names
+    BaseVoltage: nominal_voltage
+    """
+    return BaseVoltage(**locals())
+
+
+def create_connectivity_node(mrid: str = None, name: str = '', description: str = "", names: List[Name] = None, terminals: List[Terminal] = None
+                             ) -> ConnectivityNode:
+    """
+    ConnectivityNode(IdentifiedObject)
+    IdentifiedObject: mrid, name, description, names
+    ConnectivityNode: terminals
+    """
+    cn = ConnectivityNode(**locals())
+    if terminals:
+        for t in terminals:
+            t.connectivity_node = cn
+    return cn
+
+
+def create_feeder(mrid: str = None, name: str = '', description: str = "", names: List[Name] = None, location: Location = None,
+                  asset_info: AssetInfo = None, equipment: Dict[str, Equipment] = None, normal_head_terminal: Terminal = None,
+                  normal_energizing_substation: Substation = None, current_equipment: List[Equipment] = None) -> Feeder:
+    """
+    Feeder(EquipmentContainer(ConnectivityNodeContainer(PowerSystemResource(IdentifiedObject))))
+    IdentifiedObject: mrid, name, description, names
+    PowerSystemResource: location, asset_info
+    ConnectivityNodeContainer:
+    EquipmentContainer: equipment
+    Feeder: normal_head_terminal, normal_energizing_substation, current_equipment
+    """
+    f = Feeder(**locals())
+    add_equipment_container_connection(f, equipment)
+    if normal_energizing_substation:
+        normal_energizing_substation.add_feeder(f)
+    if current_equipment:
+        for equipment in current_equipment:
+            equipment.add_current_feeder(f)
+    return f
+
+
+def create_geographical_region(mrid: str = None, name: str = '', description: str = "", names: List[Name] = None,
+                               sub_geographical_regions: List[SubGeographicalRegion] = None) -> GeographicalRegion:
+    """
+    GeographicalRegion(IdentifiedObject)
+    IdentifiedObject: mrid, name, description, names
+    GeographicalRegion: sub_geographical_regions
+    """
+    gr = GeographicalRegion(**locals())
+    if sub_geographical_regions:
+        for sgr in sub_geographical_regions:
+            sgr.geographical_region = gr
+    return gr
+
+
+def create_name(name: str, type: NameType, identified_object: IdentifiedObject = None) -> Name:
+    """
+    Name()
+    Name: name, type, identified_object
+    """
+    # noinspection PyArgumentList
+    n = Name(**locals())
+    if identified_object:
+        type.get_or_add_name(n, identified_object)
+    if identified_object.name is not n.name:
+        identified_object.add_name(n)
+    return n
+
+
+def create_name_type(name: str, description: str = "", _names_index: Dict[str, Name] = dict(), _names_multi_index: Dict[str, List[Name]] = dict()) -> NameType:
```

Comment:
As above I am not sure if this has a place here or not. You create names off name types which already adds the reverse link so passing in names here makes no sense as they will either already exist or belong to a different `NameType`.